Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,213 @@
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <string>
+#include <iostream>
+#include <vector>
+#include <boost/filesystem.hpp>
+#include "MolSupplier.h"
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+namespace fs = boost::filesystem;
+
+namespace RDKit{
+	struct SupplierOption{
+		bool takeOwnership = true;
+		bool sanitize = true;
+		bool removeHs = true;
+		bool strictParsing = true;
+
+		std::string delimiter = "\t";
+		int smilesColumn = 0;
+		int nameColumn = 1;
+		bool titleLine = true;
+
+		std::string nameRecord = "";
+		int confId2D = -1;
+		int confId3D = 0;
+	};
+
+  class GeneralFileReader{
+	  public:
+			std::string fname;
```

Comment:
I generally prefer to the names of prefix member variables with `d_` so that it's easy to recognize them in code. In this case `fname` would be `d_fname`, `file_format` would be `d_fileFormat` (renamed to be consistent, we don't use underscores to separate words in variable/method/function names) and `compression_format` would be `d_compressionFormat`.